### ISSUE TITLE
After changing mutableStateOf, a wrong window is closed

### ISSUE THREAD

**Issue by igordmn** [JetBrains/compose-jb#1487]
Compose 1.0.0-rc5

```
import androidx.compose.foundation.layout.Row
import androidx.compose.material.Checkbox
import androidx.compose.material.Text
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.window.Window
import androidx.compose.ui.window.application

fun main() = application {
    var isOpen by remember { mutableStateOf(true) }

    if (isOpen) {
        Window(onCloseRequest = ::exitApplication) {}
    }

    Window(onCloseRequest = ::exitApplication) {
        Row(verticalAlignment = Alignment.CenterVertically) {
            Checkbox(isOpen, { isOpen = !isOpen })
            Text("isOpen")
        }
    }
}
```

https://user-images.githubusercontent.com/5963351/143778248-6c63d2ab-01e2-4ad0-b2c3-bf0b910ef2f6.mp4

Wrapping into `key` helps:
```
import androidx.compose.foundation.layout.Row
import androidx.compose.material.Checkbox
import androidx.compose.material.Text
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.window.Window
import androidx.compose.ui.window.application

fun main() = application {
    var isOpen by remember { mutableStateOf(true) }

    if (isOpen) {
        key(1) {
            Window(onCloseRequest = ::exitApplication) {}
        }
    }

    key(2) {
        Window(onCloseRequest = ::exitApplication) {
            Row(verticalAlignment = Alignment.CenterVertically) {
                Checkbox(isOpen, { isOpen = !isOpen })
                Text("isOpen")
            }
        }
    }
}
```

**Comment by igordmn** [comment]
Reproducer without `Window`'s:
```
import androidx.compose.material.Checkbox
import androidx.compose.material.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.DisposableEffect
import androidx.compose.runtime.LaunchedEffect
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.window.singleWindowApplication
import kotlinx.coroutines.delay

fun main() = singleWindowApplication {
    var isOpen by remember { mutableStateOf(true) }

    if (isOpen) {
        Test("test1") {
            Text("Text")
        }
    }

    Test("test2") {
        Checkbox(isOpen, { isOpen = !isOpen })
    }

    LaunchedEffect(Unit) {
        delay(1000)
        isOpen = false
        delay(1000)
    }
}

@Composable
fun Test(
    title: String = "Untitled",
    content: @Composable () -> Unit
) {
    DisposableEffect(Unit) {
        println("create $title")
        onDispose {
            println("dispose $title")
        }
    }
    content()
}
```

**Comment by igordmn** [comment]
Reproducible since `1.0.0-alpha1-rc1`.

This works:
```
import androidx.compose.foundation.layout.Row
import androidx.compose.material.Checkbox
import androidx.compose.material.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.window.Window
import androidx.compose.ui.window.application

fun main() = application {
    BugfixComposable {
        var isOpen by remember { mutableStateOf(true) }

        if (isOpen) {
            Window(onCloseRequest = ::exitApplication) {}
        }

        Window(onCloseRequest = ::exitApplication) {
            Row(verticalAlignment = Alignment.CenterVertically) {
                Checkbox(isOpen, { isOpen = !isOpen })
                Text("isOpen")
            }
        }
    }
}


@Composable
private fun BugfixComposable(content: @Composable () -> Unit) {
    content()
}
```

**Comment by igordmn** [comment]
Reverting [this commit](https://android-review.googlesource.com/c/platform/frameworks/support/+/1773989) fixes the issue, but it is part of our API

**Comment by igordmn** [comment]
Issue in Compose: https://issuetracker.google.com/issues/208304105

**Comment by igordmn** [comment]
Should be fixed in 1.0.0-rc10

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-5143">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.
